frontend/www/js/omegaup/components/common/Navbar.vue

```diff
@@ -0,0 +1,174 @@
+                href="#"><img v-bind:src="gravatarURL51"> <span class="username"
+                  v-bind:title="currentUsername">{{ currentUsername }}</span> <span class=
+                  "grader-count badge"
+                  v-show="isAdmin">…</span> <span class="caret"></span></a>
+            <ul class="dropdown-menu">
+              <li v-show="!omegaUpLockDown &amp;&amp; !inContest">
+                <a href="/profile/"><span class="glyphicon glyphicon-user"></span> {{
+                T.navViewProfile }}</a>
+              </li>
+              <li>
+                <a href="/logout/"><span class="glyphicon glyphicon-log-out"></span> {{ T.navLogOut
+                }}</a>
+              </li>
+              <hr class="dropdown-separator"
+                  v-show="isAdmin &amp;&amp; !omegaUpLockDown &amp;&amp; !inContest">
+              <li class="grader-submissions"
+                  v-show="isAdmin &amp;&amp; (!omegaUpLockDown &amp;&amp; !inContest)">
+                <a class="grader-submissions-link"
+                    href="/arena/admin/">{{ T.wordsLatestSubmissions }}</a>
+              </li>
+              <li class="grader grader-status"
+                  v-show="isAdmin &amp;&amp; !omegaUpLockDown &amp;&amp; !inContest"></li>
+              <li class="grader grader-broadcaster-sockets"
+                  v-show="isAdmin &amp;&amp; !omegaUpLockDown &amp;&amp; !inContest"></li>
+              <li class="grader grader-embedded-runner"
+                  v-show="isAdmin &amp;&amp; !omegaUpLockDown &amp;&amp; !inContest"></li>
+              <li class="grader grader-queues"
+                  v-show="isAdmin &amp;&amp; !omegaUpLockDown &amp;&amp; !inContest"></li>
+            </ul>
+          </li>
+        </ul>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop } from 'vue-property-decorator';
+import { T } from '../../omegaup.js';
+import omegaup from '../../api.js';
+import notifications_List from '../notification/List.vue';
+
+interface NavbarComponent {
+  omegaUpLockDown: boolean;
+  inContest: boolean;
+  isLoggedIn: boolean;
+  isReviewer: boolean;
+  gravatarURL51: string;
+  currentUsername: string;
+  requestURI: string;
+  isAdmin: boolean;
+  lockDownImage: string;
+}
+
+@Component({
+  components: {
+    'omegaup-notification-list': notifications_List,
+  },
+})
+export default class Navbar extends Vue {
+  @Prop() data!: NavbarComponent;
+
+  omegaUpLockDown = this.data.omegaUpLockDown;
+  inContest = this.data.inContest;
+  isLoggedIn = this.data.isLoggedIn;
+  isReviewer = this.data.isReviewer;
+  gravatarURL51 = this.data.gravatarURL51;
+  currentUsername = this.data.currentUsername;
+  requestURI = this.data.requestURI;
+  isAdmin = this.data.isAdmin;
+  lockDownImage = this.data.lockDownImage;
+  notifications: omegaup.Notification[] = [];
+  T = T;
+
+  get formattedLoginURL() {
+    return `/login/?redirect=${this.requestURI}`;
```

Comment:
ah sabes qué? `requestURI` ya no se necesita!

```suggestion
    return `/login/?redirect=${encodeURIComponent(window.location.pathname)}${encodeURIComponent(window.location.search)}`;
```

posiblemente extrayendo algunas partes para que no sea _tan_ larga.

Reply:
Creo que con la primera parte funciona correctamente